File: BOJ/CPP/단계별로_풀어보기/05_1차원_배열/BOJ_3052.cpp
/*
	BOJ 3052 : 나머지 

	author : kdgyun
	link https://st-lab.tistory.com
	link https://github.com/kdgyun
	Created by kdgyun
	Copyright © 2021 kdgyun. All rights reserved.
*/

#include <iostream>

using namespace std;

int main(int argc, const char * argv[]) {

	ios_base::sync_with_stdio(false);
	cin.tie(0);

	/*
	  0으로 초기화를 해주어야 한다.
	  bool 배열을 활용 할 경우 bool count[42] = {false} 이런식으로
	  초기화 해줄 수 있다.
	 */
	int count[42] = {};	
	
	int v;
	for(int i = 0; i < 10; i++) {
		cin >> v;
		// 입력 받은 수를 42로 나눈 나머지 인덱스의 값을 증가시킨다.
		count[v % 42]++;	
	}

	int result = 0;
	/*
	  배열을 순회하면서 적어도 한 번 이상 나온 경우에는
	  result 값을 증가시킨다. (서로 다른 수를 세기 위함)
	 */
	for(int v : count) {
		if(v > 0) {		
			result++;
		}
	}
	cout << result;

	return 0;


}

File: BOJ/Java/단계별로_풀어보기/B_04_1차원_배열/BOJ_3052.java
package 단계별로_풀어보기.B_04_1차원_배열;

/**
 * BOJ 3052 : 나머지
 * 
 * @author kdgyun 
 * {@link https://st-lab.tistory.com}
 * {@link https://github.com/kdgyun}
 *
 */

import java.io.BufferedReader;
import java.io.InputStreamReader;
import java.io.IOException;

public class BOJ_3052 {

	public static void main(String[] args) throws IOException {
		
		BufferedReader br = new BufferedReader(new InputStreamReader(System.in));

		boolean[] set = new boolean[42];
		
		for (int i = 0; i < 10; i++) {
			set[Integer.parseInt(br.readLine()) % 42] = true;
		}
		
		int count = 0;
		
		for(int i = 0; i < 42; i++) {
			if(set[i]) count++;
		}
		
		System.out.println(count);
	}
}


/* or useing hashset solution
  
 
import java.io.BufferedReader;
import java.io.InputStreamReader;
import java.io.IOException;
import java.util.HashSet;

public class BOJ_3052 {

	public static void main(String[] args) throws IOException {
		
		BufferedReader br = new BufferedReader(new InputStreamReader(System.in));

		HashSet<Integer> set = new HashSet<Integer>();
		
		for (int i = 0; i < 10; i++) {
			set.add(Integer.parseInt(br.readLine()) % 42);
		}
		
		int count = 0;

		System.out.println(set.size());
	}
}

 */
